### PR TITLE
py-gsutil: add 5.24, fix and add dependencies

### PR DESCRIPTION
**Pull request by manuelakuhn**
https://github.com/GoogleCloudPlatform/gsutil/tree/v5.24

- py-google-auth: add 2.20.0 and new variant aiohttp:
	<https://github.com/googleapis/google-auth-library-python/tree/v2.20.0>
- py-openssl: add 23.2.0
	<https://github.com/pyca/pyopenssl/tree/23.2.0>
	to prevent `AttributeError: module 'lib' has no attribute 'X509_V_FLAG_CB_ISSUER_CHECK'`
- py-httplib2: add 0.22.0
	<https://github.com/httplib2/httplib2/tree/v0.22.0>
- py-gcs-oauth2-boto-plugin: add 3.0
	<https://github.com/GoogleCloudPlatform/gcs-oauth2-boto-plugin/tree/v3.0>
- py-pyu2f: add new package
	<https://github.com/google/pyu2f/tree/v0.1.5>
- py-google-reauth: add new package
	<https://github.com/Google/google-reauth-python/tree/0.1.1>

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-gsutil/package.py [hunk 1]
```diff
@@ -11,23 +11,33 @@ class PyGsutil(PythonPackage):
...
     depends_on("python@2.7:2.8,3.5:3", type=("build", "run"))
     depends_on("py-setuptools", type="build")
+
     depends_on("py-argcomplete@1.9.4:", type=("build", "run"))
     depends_on("py-crcmod@1.7:", type=("build", "run"))
     depends_on("py-fasteners@0.14.1:", type=("build", "run"))
+    depends_on("py-gcs-oauth2-boto-plugin@3:", when="@5:", type=("build", "run"))
     depends_on("py-gcs-oauth2-boto-plugin@2.7:", type=("build", "run"))
     depends_on("py-google-apitools@0.5.32:", type=("build", "run"))
+    depends_on("py-httplib2@0.20.4:", when="@5.7:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gsutil/package.py]:
```suggestion
    depends_on("py-httplib2@0.20.4", when="@5.7:", type=("build", "run"))
```
Exact version. Didn't check older versions

File: var/spack/repos/builtin/packages/py-httplib2/package.py [hunk 2]
```diff
@@ -12,7 +12,11 @@ class PyHttplib2(PythonPackage):
...
     pypi = "httplib2/httplib2-0.13.1.tar.gz"
 
+    version("0.22.0", sha256="d7a10bc5ef5ab08322488bde8c726eeee5c8618723fdb399597ec58f3d82df81")
     version("0.18.0", sha256="b0e1f3ed76c97380fe2485bc47f25235453b40ef33ca5921bb2897e257a49c4c")
     version("0.13.1", sha256="6901c8c0ffcf721f9ce270ad86da37bc2b4d32b8802d4a9cec38274898a64044")
 
+    depends_on("py-setuptools@40.8.0:", when="@0.18.1:", type="build")
     depends_on("py-setuptools", type="build")
+
+    depends_on("py-pyparsing@2.4.2:2,3.0.4:3", when="@0.19:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-httplib2/package.py]: For these kinds of things I usually do this nowadays:
```suggestion
    depends_on("py-pyparsing@2.4.2:3", when="@0.19:", type=("build", "run"))
    conflicts("py-pyparsing@3.0.1:3.0.3", when="@0.19:")
```
But both options work.

File: var/spack/repos/builtin/packages/py-pyopenssl/package.py [hunk 3]
```diff
@@ -17,11 +17,15 @@ class PyPyopenssl(PythonPackage):
     homepage = "https://pyopenssl.org/"
     pypi = "pyOpenSSL/pyOpenSSL-19.0.0.tar.gz"
 
+    version("23.2.0", sha256="276f931f55a452e7dea69c7173e984eb2a4407ce413c918aa34b55f82f9b8bac")
     version("22.1.0", sha256="7a83b7b272dd595222d672f5ce29aa030f1fb837630ef229f62e72e395ce8968")
     version("19.0.0", sha256="aeca66338f6de19d1aa46ed634c3b9ae519a64b458f8468aec688e7e3c20f200")
 
     depends_on("py-setuptools", type="build")
+
+    depends_on("py-cryptography@38:39,40.0.2:41", when="@23.2:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyopenssl/package.py]: Same optional comment here

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-gsutil/package.py (hunk 1 above)
Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/py-gsutil/package.py]: Ah thanks, they pinned it down in 5.17 which I overlooked.

File: var/spack/repos/builtin/packages/py-httplib2/package.py [hunk 4]
```diff
@@ -12,7 +12,13 @@ class PyHttplib2(PythonPackage):
...
+    version("0.22.0", sha256="d7a10bc5ef5ab08322488bde8c726eeee5c8618723fdb399597ec58f3d82df81")
+    version("0.20.4", sha256="58a98e45b4b1a48273073f905d2961666ecf0fbac4250ea5b47aef259eb5c585")
     version("0.18.0", sha256="b0e1f3ed76c97380fe2485bc47f25235453b40ef33ca5921bb2897e257a49c4c")
     version("0.13.1", sha256="6901c8c0ffcf721f9ce270ad86da37bc2b4d32b8802d4a9cec38274898a64044")
 
+    depends_on("py-setuptools@40.8.0:", when="@0.18.1:", type="build")
     depends_on("py-setuptools", type="build")
+
+    depends_on("py-pyparsing@2.4.2:3", when="@0.19:", type=("build", "run"))
+    conflicts("py-pyparsing@3.0.1:3.0.3", when="@0.19:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-httplib2/package.py]: Actually I think it needs to be
```suggestion
    conflicts("^py-pyparsing@3.0.1:3.0.3", when="@0.19:")
```

**Review by adamjstewart — APPROVED** [review]

**Comment by manuelakuhn** [comment]
@spackbot run pipeline

**Pull request merged**